### PR TITLE
ARTEMIS-756 Moving Client CDI as a main module

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
@johnament  can you take a look...

The only commit here is ARTEMIS-756 Moving Client CDI. The other two are already part of master, I'm not sure why github is showing 3 commits

File: artemis-rest/src/main/java/org/apache/activemq/artemis/rest/Jms.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import javax.ws.rs.ext.Providers;
+import java.io.ByteArrayInputStream;
+import java.lang.reflect.Type;
+
+import org.apache.activemq.artemis.rest.util.HttpMessageHelper;
+import org.jboss.resteasy.core.Headers;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.util.GenericType;
+
+public class Jms {
```
Comment (johnament) [inline artemis-rest/src/main/java/org/apache/activemq/artemis/rest/Jms.java]: Is this on purpose?

File: artemis-rest/src/main/java/org/apache/activemq/artemis/rest/integration/EmbeddedRestActiveMQ.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ protected void initEmbeddedActiveMQ() {
       embeddedActiveMQ = new EmbeddedActiveMQ();
    }
 
-   MessageServiceManager getManager() {
+   public MessageServiceManager getManager() {
```
Comment (johnament) [inline artemis-rest/src/main/java/org/apache/activemq/artemis/rest/integration/EmbeddedRestActiveMQ.java]: Same...

File: artemis-rest/src/main/java/org/apache/activemq/artemis/rest/integration/EmbeddedRestActiveMQJMS.java [hunk 3]
```diff
@@ -20,9 +20,9 @@
 import org.apache.activemq.artemis.jms.server.embedded.EmbeddedJMS;
 import org.apache.activemq.artemis.spi.core.naming.BindingRegistry;
 
-class EmbeddedRestActiveMQJMS extends EmbeddedRestActiveMQ {
+public class EmbeddedRestActiveMQJMS extends EmbeddedRestActiveMQ {
```
Comment (johnament) [inline artemis-rest/src/main/java/org/apache/activemq/artemis/rest/integration/EmbeddedRestActiveMQJMS.java]: Same...

File: artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/Embedded.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.activemq.artemis.core.config.impl.ConfigurationImpl;
+import org.apache.activemq.artemis.core.remoting.impl.invm.InVMAcceptorFactory;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.ActiveMQServers;
+import org.apache.activemq.artemis.rest.MessageServiceConfiguration;
+import org.apache.activemq.artemis.rest.MessageServiceManager;
+import org.jboss.resteasy.plugins.server.tjws.TJWSEmbeddedJaxrsServer;
+import org.jboss.resteasy.test.TestPortProvider;
+
+public class Embedded {
```
Comment (johnament) [inline artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/Embedded.java]: Same...

File: artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/EmbeddedTest.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+import org.jboss.resteasy.client.ClientRequest;
+import org.jboss.resteasy.client.ClientResponse;
+import org.jboss.resteasy.spi.Link;
+import org.jboss.resteasy.test.TestPortProvider;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class EmbeddedTest {
```
Comment (johnament) [inline artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/EmbeddedTest.java]: Same...

File: artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/JMSTest.java [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+import org.jboss.resteasy.client.ClientRequest;
+import org.jboss.resteasy.client.ClientResponse;
+import org.jboss.resteasy.spi.Link;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.jboss.resteasy.test.TestPortProvider.generateURL;
+
+public class JMSTest extends MessageTestBase {
```
Comment (johnament) [inline artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/JMSTest.java]: SAme...

File: artemis-rest/src/main/java/org/apache/activemq/artemis/rest/Jms.java (hunk 1 above)
Comment (clebertsuconic) [inline artemis-rest/src/main/java/org/apache/activemq/artemis/rest/Jms.java]: it is not part of this PR.. github is acting up.

File: artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/JMSTest.java (hunk 6 above)
Comment (clebertsuconic) [inline artemis-rest/src/test/java/org/apache/activemq/artemis/rest/test/JMSTest.java]: @johnament github is acting up... you should look at the commits...  there are not many changes.. basically moving stuff.

**Comment by johnament** [comment]
I'm fine with the CDI stuff going in.
